XV/Assets/Scripts/Utils/CameraHandler.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class CameraHandler : MonoBehaviour {
+
+	public enum Mode { FREE, SUBJECTIVE, LOCKED }
+
+	[SerializeField] private float StandardSpeed;
+	[SerializeField] private float FastSpeedMultiplier;
+    [SerializeField] private float MouseSensitivity;
+	[SerializeField] private Text CameraModeText;
+
+	private bool mIsRepositioning;
+
+	private Mode mMode;
+	public Mode CameraMode
+	{
+		get
+		{
+			return mMode;
+		}
+		set
+		{
+			if (value != CameraMode) {
+				switch (value)
+				{
+					case Mode.LOCKED:
+						SetLockedMode();
+						break;
+					case Mode.FREE:
+						SetFreeMode();
+						break;
+					case Mode.SUBJECTIVE:
+						SetSubjectiveMode();
+						break;
+				}
+				CameraModeText.text = "Camera: " + value.ToString();
+			}
+		}
+	}
+
+	private void Start()
+	{
+		CameraMode = Mode.LOCKED;
+		mIsRepositioning = false;
+	}
+
+    void Update()
+    {
+		// Set camera modes using 1 / 2 / 3 keys
+		if (Input.GetKeyDown(KeyCode.Alpha1)) {
+			CameraMode = Mode.LOCKED;
+		}
+		else if (Input.GetKeyDown(KeyCode.Alpha2)) {
+			CameraMode = Mode.FREE;
+		}
+		else if (Input.GetKeyDown(KeyCode.Alpha3)) {
+			CameraMode = Mode.SUBJECTIVE;
+		}
+		// No movement if in locked mode
+		if (CameraMode != Mode.LOCKED)
+		{
+			if (!mIsRepositioning) {
+				ApplyMovement();
+			}
+			ApplyRotation();
+		}
+    }
+
+	private void ApplyMovement()
+	{
+		bool fastMode = Input.GetKey(KeyCode.LeftShift);
```

Comment:
Var local = lVar
Var parameter = iVar
Var to return = oVar

Var attribut = mVar
static Var  = sVar

enum = VAR
const = VAR